metriken/src/dynmetrics.rs

```diff
@@ -64,6 +64,7 @@ pub struct MetricBuilder {
     name: Cow<'static, str>,
     desc: Option<Cow<'static, str>>,
     metadata: HashMap<String, String>,
+    formatter: &'static dyn Fn(&MetricEntry, Format) -> String,
```

Comment:
As a suggestion, this should probably just be a raw function pointer
```suggestion
    formatter: fn(&MetricEntry, Format) -> String,
```

I don't think you're ever going to use this with something that captures variables and, if you did, it would be impossible to use since you need a `'static` reference anyway.

As a bonus this means you can do `#[metric(formatter = x)]` without having to take a reference.